### ISSUE TITLE
Missing types for `category.supportedModes` and `node.explanation`

### ISSUE THREAD

**Issue by hilja** [GoogleChrome/lighthouse#15799]
### FAQ

- [X] Yes, my issue is not about [variability](https://github.com/GoogleChrome/lighthouse/blob/main/docs/variability.md) or [throttling](https://github.com/GoogleChrome/lighthouse/blob/main/docs/throttling.md).
- [X] Yes, my issue is not about a specific accessibility audit (file with [axe-core](https://github.com/dequelabs/axe-core) instead).

### URL

https://clubmate.fi/about

### What happened?

Not a big deal, but type defs for `category.supportedModes` and `node.explanation` fields seem to be missing. Also for www.theverge.com the `node.explanation` is not there (under the `label-content-name-mismatch` item). Apparently these are more rare fields.

Is this a mistake in the types or the output?

### What did you expect?

Complete typing.

### What have you tried?

I saved the output to a `.ts` file and added a type, then looked for the red squiglies:

```ts
import type Result from 'lighthouse/types/lhr/lhr.js'

const foo: Result = { /* LH output */ }
```

<img width="795" alt="Screenshot 2024-02-02 at 19 21 55" src="https://github.com/GoogleChrome/lighthouse/assets/2558021/bf1cde98-4d7f-4c82-8f84-d8eba7e3d978">

<img width="810" alt="Screenshot 2024-02-02 at 20 02 14" src="https://github.com/GoogleChrome/lighthouse/assets/2558021/fac12769-b99e-437b-8a21-c17e90318550">

### How were you running Lighthouse?

node

### Lighthouse Version

11.5.0

### Node Version

v20.11.0

### OS

Mac OS

**Comment by hilja** [comment]
Okay interesting. Could you elaborate on the `Config` a bit? Or point me to docs. I don't fully understand how to use that, or why am I seeing `supportedModes` in the results?

If I use `Config.Category`:

```ts
type CategoryConfig = Record<string, Config.Category>
```

Then it’s missing other properties like `id` and `score`.

I can extend the types and it seems to work:

```ts
type CategoryConfig = Record<string, Config.Category & Result.Category>
```

Is this correct usage?

Thanks!

**Comment by adamraine** [comment]
> Okay interesting. Could you elaborate on the Config a bit? Or point me to docs. I don't fully understand how to use that, or why am I seeing supportedModes in the results?

Yeah I took a look at the generated results, and looks like we are emitting `supportedModes` on the result category even though it's not a part of the type. So I guess we should either remove that from the output or add `supportedModes` to the types.

**Comment by An-Yay** [comment]
I'd like to have a take on this issue

**Comment by adamraine** [comment]
Feel free to submit a PR @An-Yay

**Comment by hilja** [comment]
Please do it :) I just patched Lighthouse and forgot about it.

**Comment by parshva-b** [comment]
Hey @adamraine,

I can pick the issue if it is still pending!

**Comment by adamraine** [comment]
Feel free to submit a PR @parshva-b

**Comment by angad-sethi** [comment]
G'day folks, I've raised a PR [here](https://github.com/GoogleChrome/lighthouse/pull/16006) for this fix. Please let me know if I've got it wrong. Cheers!
